app/models/patient.rb

```diff
@@ -142,6 +166,24 @@ def self.contacted_since(datetime)
       first_contacts: first_contact.length, pledges_sent: 20 }
   end
 
+  def self.to_csv
+    CSV.generate(encoding: 'utf-8') do |csv|
+      csv << CSV_EXPORT_FIELDS.keys # Header line
+      all.each do |patient|
+        csv << CSV_EXPORT_FIELDS.values.map do |field|
+          value = patient.public_send(field)
```

Comment:
Oh, I should have mentioned that I tried that based on your earlier suggestion but it didn't seem to work for 'has_alt_contact?'